### PR TITLE
Add useUpsert hook

### PR DESCRIPTION
**Pull request by kachar**
# Motivation

Created a hook that will use the functionality exposed by `supabase-js` for doing insert or update queries via `upsert()` method.

# Related

- Discussion https://github.com/tmm/react-supabase/discussions/5
- Docs https://supabase.io/docs/reference/javascript/upsert

# Docs

[/docs/pages/documentation/data/use-upsert.md](https://github.com/kachar/react-supabase/blob/ba070f8b6f9fa220c7e05d79952657d4b4cf21bc/docs/pages/documentation/data/use-upsert.md)

### CODE REVIEW

**Comment by kachar** [comment]
Not sure why, but the docs title is not well rendered as `useUpsert`, but `Use-Upsert` instead.

image

It will be nice to fix that prior a merge

File: docs/pages/documentation/data/use-upsert.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+```
+
+## Notes
+
+- By specifying the `onConflict` option, you can make `UPSERT` work on a column(s) that has a `UNIQUE` constraint.
+- Primary keys should to be included in the data payload in order for an update to work correctly.
+- Primary keys must be natural, not surrogate. There are however, workarounds for surrogate primary keys.
+- Param `filter` makes sense only when operation is update
+- Upsert supports sending array of elements, just like `useInsert`
```
Comment (tmm) [inline docs/pages/documentation/data/use-upsert.md:29]: Thanks for adding these

File: src/hooks/data/use-upsert.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    count?: number | null
+    data?: Data | Data[] | null
+    error?: PostgrestError | null
+    fetching: boolean
+}
+
+export type UseUpsertResponse<Data = any> = [
+    UseUpsertState<Data>,
+    (
+        values: Partial<Data>,
```
Comment (tmm) [inline src/hooks/data/use-upsert.ts:17]: Let's have this match `execute`'s signature.

```suggestion
        values: Partial<Data> | Partial<Data>[],
```

**Review by tmm — CHANGES_REQUESTED** [review]
Looks good. One small thing, then I'll merge and release.

File: src/hooks/data/use-upsert.ts (hunk 2 above)
Comment (kachar) [reply src/hooks/data/use-upsert.ts]: Awesome finding :raised_hands: I haven't noticed that one

**Pull request merged**